ffi/src/lib.rs

```diff
@@ -730,6 +730,61 @@ pub unsafe extern "C" fn fwd_free_value(value: *const Value) {
     }
 }
 
+/// Struct returned by `fwd_create_db` and `fwd_open_db`
+#[derive(Debug)]
+#[repr(C)]
+pub struct DatabaseCreationResult {
+    pub db: *const DatabaseHandle<'static>,
```

Comment:
This doesn't need to be behind a pointer. cbindgen will do the right thing of you do `Option<Box<DatabaseHandle<'static>>>` instead. This applies to all structs fully known to cbindgen and also works on functions. So, instead of `ptr: *mut DatabaseHandle<'static>`, we can do `mut hdl: Box<DatabaseHandle<'static>>` to reduce some unsafe code. It will also handle `&DatabaseHandle<'static>` and `&mut DatabaseHandle<'static>` correctly.

```suggestion
    pub db: Option<Box<DatabaseHandle<'static>>>,
```

Reply:
This will be address with #967